### PR TITLE
Improve handling of annotation processor in eclipse

### PR DESCRIPTION
**Pull request by tobiasdiez**
I played around a bit with JabRef in VS Code, and encountered a problem with the annotation processing. It was quite easy to fix by putting the files generated by the annotation processor in the `src/gen` folder. 

Since VS Code uses the same libraries as eclipse under the hood, this fix should also apply to eclipse directly. @Siedlerchr can you please confirm. Thanks!

Background infos: http://dplatz.de/blog/2018/gradle-apt.html

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Unfortunately it did not work on my windows machine

**Comment by tobiasdiez** [comment]
Does the `Log4jPlugins.java` exist in `src/main/gen` ? Are there error messages?

**Comment by Siedlerchr** [comment]
I tried again. I executed gradlew run first. Then I can run it in Eclipse as well. Maybe the processResources command needs to be executed for the eclipse command as well.

File: docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace.md [hunk 1]
```diff
@@ -189,16 +189,6 @@ Contributions to JabRef's source code need to have a code formatting that is con
 
       ![checkstyle settings](../.gitbook/assets/intellij-checkstyle-settings.png)
 
-#### Troubleshooting when using both IDEA and Eclipse
-
-If you have configured Eclipse for the same project \(the required steps are described below\), then the additionally added file `Log4jPlugins.java` must be excluded from the compilation process, otherwise an error will occur during the compilation of the project:
```
Comment (koppor) [inline docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace.md:194]: I still need this hack in IntelliJ 2020.1

Comment (koppor) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace.md:194]: Addressed in https://github.com/JabRef/jabref/pull/6794.

**Pull request merged**